### PR TITLE
update mace wrapper to enable training

### PR DESCRIPTION
**Pull request by ys-teh**
# ALCHEMI Toolkit Pull Request

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Changes Made

- Changed training=False to training=self.training in MACEWrapper using the PyTorch's flag to be compatible with the upcoming model training workflow.
- Changed mock model used for MACE in unit test to be dependent on a trainable parameter so train mode test makes sense.
- Added a unit test to make sure that optimization now works when using MACEWrapper.

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [ ] I have performed a self-review of my code
- [ ] I have added docstrings to new functions/classes
- [ ] I have updated the documentation (if applicable)

## Additional Notes

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

**Review by dallasfoster — APPROVED** [review]

**Comment by dallasfoster** [comment]
/ok to test [1397531](https://github.com/NVIDIA/nvalchemi-toolkit/pull/97/commits/13975311b9c1a9575d43c96a204b580410797084)

**Review by laserkelvin — APPROVED** [review]
LGTM
